concordium-consensus/tests/scheduler/SchedulerTests/SmartContracts/V1/Queries.hs

```diff
@@ -1,614 +1,724 @@
+                TJSON
                     { payload = InitContract initAmount V1 contractBalanceTransferSourceFile "init_contract" "",
-                      metadata = makeDummyHeader alesAccount 2 100000,
-                      keys = [(0, [(0, alesKP)])]
-                    },
-                  (SuccessWithSummary (initializationCostCheck contractBalanceTransferSourceFile "init_contract"), emptySpec)
-                ),
-                ( TJSON
+                      metadata = makeDummyHeader accountAddress0 2 100_000,
+                      keys = [(0, [(0, keyPair0)])]
+                    },
+              taaAssertion = \result _ ->
+                return $ Helpers.assertSuccess result
+            },
+          Helpers.TransactionAndAssertion
+            { taaTransaction =
+                TJSON
                     { payload = Update updateAmount (Types.ContractAddress 0 0) "contract.query" parameters,
-                      metadata = makeDummyHeader alesAccount 3 100000,
-                      keys = [(0, [(0, alesKP)])]
-                    },
-                  (SuccessWithSummary (successWithEventsCheck eventsCheck), emptySpec)
-                )
-            ]
-        }
-  where
+                      metadata = makeDummyHeader accountAddress0 3 100_000,
+                      keys = [(0, [(0, keyPair0)])]
+                    },
+              taaAssertion = \result _ ->
+                -- Check the number of events:
+                -- - 3 events for transfering
+                -- - 1 event for a succesful update to the contract.
+                return $ Helpers.assertSuccessWhere (Helpers.assertNumberOfEvents 4) result
+            }
+        ]
     initAmount = 123
     updateAmount = 456
     transferAmount = 78
 
     parameters = BSS.toShort $ runPut $ do
-        put thomasAccount
+        put accountAddress1
         Wasm.putAmountLE transferAmount
         putWord64le 0 -- Index of the contract address.
         putWord64le 0 -- Subindex of the contract address.
         Wasm.putAmountLE $ initAmount + updateAmount - transferAmount -- Expected balance.
-    eventsCheck :: [Types.Event] -> Expectation
-    eventsCheck events = do
-        -- Check the number of events:
-        -- - 3 events for transfering
-        -- - 1 event for a succesful update to the contract.
-        eventsLengthCheck 4 events
 
 contractBalanceMissingContractSourceFile :: FilePath
 contractBalanceMissingContractSourceFile = "./testdata/contracts/v1/queries-contract-balance-missing-contract.wasm"
 
-contractBalanceMissingContractTestCase :: TestCase PV5
-contractBalanceMissingContractTestCase =
-    TestCase
-        { tcName = "Query the balance of a missing contract",
-          tcParameters = (defaultParams @PV5){tpInitialBlockState = initialBlockState},
-          tcTransactions =
-            [   ( TJSON
+contractBalanceMissingContractTestCase ::
+    forall pv.
+    Types.IsProtocolVersion pv =>
+    Types.SProtocolVersion pv ->
```

Comment:
Why are all these written with a generic `pv` when they only apply to `pv = P5`?

Reply:
They also apply to `pv = P6` now, which also seems reasonable, since it will run against the new consensus